heat/core/dndarray.py

```diff
@@ -566,6 +566,54 @@ def __cast(self, cast_function) -> Union[float, int]:
 
         raise TypeError("only size-1 arrays can be converted to Python scalars")
 
+    def collect_(self, target_rank: Optional[int] = 0) -> None:
+        """
+        A method collecting a distributed tensor to one rank, chosen by the target_rnk variable, and zero by default.
+        It is a specific case of the redistribute_ method.
+
+        Parameters
+        ----------
+        target_rank : int, optional
+            The rank to which the tensor will be collected.
+
+        Raises
+        ------
+        TypeError
+            If the target rank is not an integer.
+        ValueError
+            If the target rank is out of bounds.
+
+        Examples
+        --------
+        >>> a = st = ht.ones((50, 81, 67), split=2)
+        >>> print(st.lshape)
+        [0/2] (50, 81, 23)
+        [1/2] (50, 81, 22)
+        [2/2] (50, 81, 22)
+        >>> st.collect_()
+        >>> print(st.lshape)
+        [0/2] (50, 81, 67)
+        [1/2] (50, 81, 0)
+        [2/2] (50, 81, 0)
+        >>> st.collect_(1)
+        >>> print(st.lshape)
+        [0/2] (50, 81, 0)
+        [1/2] (50, 81, 67)
+        [2/2] (50, 81, 0)
+        """
+        if not isinstance(target_rank, int):
+            raise TypeError(f"target rank must be of type int , but was {type(target_rank)}")
+        if target_rank >= self.comm.size:
+            raise ValueError("target rank is out of bounds")
+        if not self.is_distributed():
+            return
+
+        target_map = self.lshape_map.clone()
+        for i in range(self.comm.size):
+            # branchless way of putting 0 in all entries except for the target rank
+            target_map[i, self.split] = self.gshape[self.split] * (i == target_rank)
```

Comment:
Better vectorize whenever possible. Something like this?

```suggestion
        target_map[:, self.split] = 0
        target_map[target_rank, self.split] = self.gshape[self.split]
```